bin/xbps-reconfigure/main.c

```diff
@@ -176,10 +187,43 @@ main(int argc, char **argv)
 		rv = xbps_configure_packages(&xh, ignpkgs);
 	} else {
 		for (i = optind; i < argc; i++) {
-			rv = xbps_configure_pkg(&xh, argv[i], true, false);
+			char* pkg = argv[i];
+			if (rdeps) {
+				rv = find_pkg_deps(&xh, pkg, fulldeptree, &deps);
+				if (rv != 0) {
+					fprintf(stderr, "Failed to collect dependencies for "
+						"`%s': %s\n", pkg, strerror(rv));
+				}
+				for (unsigned int j = 0; j < xbps_array_count(deps); j++) {
+					const char *pkgdep = NULL;
+					char pkgname[XBPS_NAME_SIZE];
+					xbps_array_get_cstring_nocopy(deps, j, &pkgdep);
+
+					if (fulldeptree) {
+						if (!xbps_pkg_name(pkgname, sizeof(pkgname), pkgdep)) {
+							fprintf(stderr,
+								"Unable to get package name for dependency `%s'\n", pkgdep);
+							exit(EXIT_FAILURE);
+						}
+					} else {
+						if (!xbps_pkgpattern_name(pkgname, sizeof(pkgname), pkgdep)) {
+							fprintf(stderr,
+								"Unable to get package name for dependency `%s'\n", pkgdep);
+							exit(EXIT_FAILURE);
```

Comment:
Hmm can this function actually fail? :/

Reply:
probably not in this usage, but it has similar error handling when used in `xbps-uhelper`

Reply:
Hmm I think I'd either remove the error checking entirely or not quit if it happens; leave to someone else to make a final decision on that though.